firmware/baseband/proc_btlerx.cpp

```diff
@@ -26,245 +27,476 @@
-                    bool current_bit;
-                    uint8_t byte = 0;
-                    for (int c = 0; c < 8; c++) {
-                        if (rb_buf[(rb_head + 5 * 8 + t * 8 + c) % RB_SIZE] > g_threshold)
-                            current_bit = true;
-                        else
-                            current_bit = false;
-                        byte |= current_bit << (7 - c);
-                    }
-
-                    packet_data[t] = byte;
-                }
+        for (j = 0; j < SAMPLE_PER_SYMBOL; j++) {
+            // Sample and compare with the adjascent next sample.
+            I0 = dst_buffer.p[i + j].real();
+            Q0 = dst_buffer.p[i + j].imag();
+            I1 = dst_buffer.p[i + j + 1].real();
+            Q1 = dst_buffer.p[i + j + 1].imag();
 
-                uint8_t byte_temp4 = (uint8_t)(((channel_number * 0x0802LU & 0x22110LU) | (channel_number * 0x8020LU & 0x88440LU)) * 0x10101LU >> 16);
-                uint8_t lfsr_2 = byte_temp4 | 2;
-                int pdu_crc_length = packet_length + 2 + 3;
-                int pdu_crc_counter = 0;
-                while (pdu_crc_length--) {
-                    for (uint8_t i = 0x80; i; i >>= 1) {
-                        if (lfsr_2 & 0x80) {
-                            lfsr_2 ^= 0x11;
-                            (packet_data[pdu_crc_counter]) ^= i;
-                        }
-                        lfsr_2 <<= 1;
-                    }
-                    pdu_crc_counter = pdu_crc_counter + 1;
-                }
+            phase_idx = j;
 
-                if (packet_addr_l == 0x8E89BED6) {
-                    crc[0] = crc[1] = crc[2] = 0x55;
-                } else {
-                    crc[0] = crc[1] = crc[2] = 0;
-                }
+            demod_buf_access[phase_idx][demod_buf_offset] = (I0 * Q1 - I1 * Q0) > 0 ? 1 : 0;
+
+            k = sp;
+            unequal_flag = false;
+
+            accesssAddress = 0;
 
-                uint8_t v, t, d, crc_length;
-                uint32_t crc_result = 0;
-                crc_length = packet_length + 2;
-                int counter = 0;
-                while (crc_length--) {
-                    uint8_t byte_temp5 = (uint8_t)(((packet_data[counter] * 0x0802LU & 0x22110LU) | (packet_data[counter] * 0x8020LU & 0x88440LU)) * 0x10101LU >> 16);
-                    d = byte_temp5;
-                    for (v = 0; v < 8; v++, d >>= 1) {
-                        t = crc[0] >> 7;
-                        crc[0] <<= 1;
-                        if (crc[1] & 0x80) crc[0] |= 1;
-                        crc[1] <<= 1;
-                        if (crc[2] & 0x80) crc[1] |= 1;
-                        crc[2] <<= 1;
-                        if (t != (d & 1)) {
-                            crc[2] ^= 0x5B;
-                            crc[1] ^= 0x06;
-                        }
-                    }
-                    counter = counter + 1;
+            for (p = 0; p < demod_buf_len; p++) {
```

Comment:
What's this doing? (comment)

Reply:
If we are looping through each of the bits in our access address and we found one doesnt match what we expect we break and keep looking. A lot of the demod stuff was borrowed so I had to try and understand all of this as I ported it over. Maybe theres a better way to rework this?